pyproject.toml

```diff
@@ -1,3 +1,150 @@
+
+[tool.poetry.group.qcut]
+optional = true
+
+[tool.poetry.group.qchem]
+optional = true
+
+[tool.poetry.group.data]
+optional = true
+
+[tool.poetry.group.dev.dependencies]
+pre-commit = ">=2.19.0"
+pytest = ">=7.1.2"
+pytest-cov = ">=3.0.0"
+pytest-mock = ">=3.7.0"
+pytest-xdist = ">=2.5.0"
+pytest-split = "^0.8"
+flaky = ">=3.7.0"
+pytest-forked = ">=1.4.0"
+pytest-benchmark = ">=4.0.0"
+black = "~23.12"
+tomli = "~2.0.0" # Drop once minimum Python version is 3.11
+
+[tool.poetry.group.ci.dependencies]
+cvxpy = "^1.4.1"
+cvxopt = [
+    { python = "<3.12", version = "^1.3.2" },
+    { python = "3.12", platform = "linux", url = "https://files.pythonhosted.org/packages/c7/17/ee82c745c5bda340a4dd812652c42fb71efd45f663554a10c3ec45f230df/cvxopt-1.3.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl" },
+    { python = "3.12", markers = "sys_platform == 'darwin' and platform_machine == 'x86_64'", url = "https://files.pythonhosted.org/packages/10/dc/1c21715e1267ca29f562e4450426d1ff8a7ffcc3e670100cec332a105b95/cvxopt-1.3.2-cp312-cp312-macosx_10_9_x86_64.whl" },
+    { python = "3.12", markers = "sys_platform == 'darwin' and platform_machine == 'arm64'", url = "https://files.pythonhosted.org/packages/cd/c8/a04048143d0329ccd36403951746c1a6b5f1fc56c479e5a0a77efb2064b2/cvxopt-1.3.2-cp312-cp312-macosx_11_0_arm64.whl" },
+    { python = "3.12", platform = "windows", url = "https://files.pythonhosted.org/packages/9f/ad/edce467c24529c536fc9de787546a1c8eca293009383a872b6f638d22eae/cvxopt-1.3.2-cp312-cp312-win_amd64.whl" },
+]
+matplotlib = "^3.5.0"
+
+[tool.poetry.group.doc.dependencies]
+mistune = "0.8.4"
+m2r2 = "^0.3.2"
+pygments-github-lexers = "^0.0.5"
+semantic-version = "2.10"
+docutils = "0.16"
+sphinx = "~3.5.0"  # must only provide one to export, but should be 4.2.0 for python >= 3.10
+sphinx-automodapi = "0.13"
+sphinx-copybutton = "^0.5.2"
+sphinxcontrib-bibtex = "2.4.2"
+sphinxcontrib-applehelp = "1.0.4"
+sphinxcontrib-devhelp = "1.0.2"
+sphinxcontrib-qthelp = "1.0.3"
+sphinxcontrib-htmlhelp = "2.0.1"
+sphinxcontrib-serializinghtml = "1.1.5"
+tensornetwork = "0.3"
+jinja2 = "3.0.3"
+# we do not pin the sphinx theme, to allow the
+# navbar and header data to be updated at the source
+pennylane-sphinx-theme = "*"
+sphinxext-opengraph = "0.6.3" # Note: latest version 0.9.0 requires Sphinx >=4.0
+matplotlib = "*"
+
+[tool.poetry.group.jax.dependencies]
+jax = "^0.4.14"
+jaxlib = "^0.4.14"
+
+[tool.poetry.group.torch.dependencies]
+torch = [
+    { version = "2.1.0", source = "PyPI", platform = "darwin" },
+    { version = "2.1.0+cpu", source = "pytorch-cpu", platform = "!=darwin" },
+]
+
+[tool.poetry.group.external.dependencies]
+pyzx = "^0.7.1"
+matplotlib = "*"
+stim = "*"
+
+[tool.poetry.group.qcut.dependencies]
+kahypar = "^1.1.7"
+opt_einsum = "*"
+
+[tool.poetry.group.qchem.dependencies]
+openfermionpyscf = "*"
+basis-set-exchange = "*"
+
+[tool.poetry.group.data.dependencies]
+h5py = "*"
+aiohttp = "*"
+fsspec = "*"
```

Comment:
Is there a standard (aka non-poetry) way to handle this information in the pyproject.toml file?

Reply:
from what I see, no. PEP517 and more importantly here, PEP518 are not very clear on the matter. It seems that 3P orgs can define their own 517-compliant build backend, then provide it, and that's exactly what poetry does.